src/main/java/uk/gov/ida/verifymatchingservicetesttool/scenarios/UserAccountCreationScenario.java

```diff
@@ -1,44 +1,42 @@
 package uk.gov.ida.verifymatchingservicetesttool.scenarios;
 
-import org.hamcrest.core.Is;
-import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
-import uk.gov.ida.verifymatchingservicetesttool.configurations.ConfigurationReader;
-import uk.gov.ida.verifymatchingservicetesttool.utils.FileUtils;
+import org.junit.jupiter.api.extension.ExtendWith;
+import uk.gov.ida.verifymatchingservicetesttool.configurations.ApplicationConfiguration;
+import uk.gov.ida.verifymatchingservicetesttool.resolvers.ApplicationConfigurationResolver;
 
-import javax.validation.constraints.AssertFalse;
-import javax.validation.constraints.AssertTrue;
 import javax.ws.rs.client.ClientBuilder;
 import javax.ws.rs.client.Entity;
 import javax.ws.rs.core.GenericType;
 import javax.ws.rs.core.Response;
-import java.net.URI;
 import java.util.HashSet;
 import java.util.Map;
 
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
 import static org.hamcrest.CoreMatchers.anyOf;
 import static org.hamcrest.CoreMatchers.is;
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.junit.jupiter.api.Assumptions.assumeTrue;
-import static org.junit.jupiter.api.Assumptions.assumingThat;
 
+@ExtendWith(ApplicationConfigurationResolver.class)
+public class UserAccountCreationScenario extends ScenarioBase {
 
-public class UserAccountCreationScenario {
-    private static URI accountCreationUrl = ConfigurationReader.getConfiguration().getLocalMatchingServiceAccountCreationUrl();
-    private FileUtils fileUtils = new FileUtils();
+    public UserAccountCreationScenario(ApplicationConfiguration configuration) {
+        super(configuration);
+    }
 
     @Test
     public void runForUserAccountCreation() {
-        assumingThat(accountCreationUrl != null, () -> {
-            Response response = ClientBuilder.newClient()
-                    .target(accountCreationUrl)
-                    .request("application/json")
-                    .post(Entity.json(fileUtils.readFromResources("user-account-creation.json")));
+        if (configuration.getLocalMatchingServiceAccountCreationUrl() != null) {
+            Response response = client.target(configuration.getLocalMatchingServiceAccountCreationUrl())
+                .request(APPLICATION_JSON)
+                .post(Entity.json(fileUtils.readFromResources("user-account-creation.json")));
 
             Map<String, String> result = response.readEntity(new GenericType<Map<String, String>>() {{ }});
 
-            assertThat(result.keySet(), Is.is(new HashSet<String>() {{ add("result"); }}));
+            assertThat(result.keySet(), is(new HashSet<String>() {{ add("result"); }}));
             assertThat(result.get("result"), anyOf(is("success"), is("failure")));
-        });
+        } else {
+            System.out.println("Account creation not enabled.");
+        }
```

Comment:
It looks like we could use assumeTrue to avoid this if/else clause, and mean that the test will show up as aborted if there is no user account creation url instead of successful (which seems more true...). Ought to be able to show a this message as well, although I haven't tested that. 
http://junit.org/junit5/docs/current/api/org/junit/jupiter/api/Assumptions.html

Reply:
Makes sense. Thank you.